java-checks/src/main/java/org/sonar/java/checks/ParameterReassignedToCheck.java

```diff
@@ -43,60 +46,83 @@
   private final Set<Symbol> variables = Sets.newHashSet();
 
   private JavaFileScannerContext context;
-  private SemanticModel semanticModel;
 
   @Override
   public void scanFile(final JavaFileScannerContext context) {
     this.context = context;
     variables.clear();
-    semanticModel = (SemanticModel) context.getSemanticModel();
-    scan(context.getTree());
-  }
-
-  private boolean hasSemanticModel() {
-    return semanticModel != null;
+    if (context.getSemanticModel() != null) {
+      scan(context.getTree());
+    }
   }
 
   @Override
   public void visitMethod(MethodTree tree) {
+    BlockTree block = tree.block();
+    if(block == null) {
+      return;
+    }
+    CFG cfg = CFG.build(tree);
+    LiveVariables analyze = LiveVariables.analyze(cfg);
+    Set<Symbol> live = analyze.getIn(cfg.entry());
     for (VariableTree parameterTree : tree.parameters()) {
```

Comment:
Why not rely on stream?
```
tree.parameters().stream()
  .map(VariableTree::symbol)
  .filter(symbol -> !live.contains(symbol))
  .forEach(variables::add)
```

Reply:
I see no added value for stream here... neither from readability nor perf...

Reply:
Okay, I was simply wondering about calling 4 times `parameterTree.symbol()`